File: spaceship_crm/java8/spaceship_crm_java8_jsp_jpa/src/main/java/com/spaceship/crm/repository/UserRepository.java
/**
 * Author veni.vidi.dev (veni.vidi.dev@gmail.com)
 */
package com.spaceship.crm.repository;

import com.spaceship.crm.entity.User;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserRepository extends CrudRepository<User, Long> {
    Optional<User> findByUsername(String username);
}

File: spaceship_crm/kotlin/spaceship_crm_kotlin_spring_mustache_jpa/src/main/kotlin/com/spaceship/crm/repository/UserRepository.kt
/**
 * Author veni.vidi.dev (veni.vidi.dev@gmail.com)
 */
package com.spaceship.crm.repository

import org.springframework.data.repository.CrudRepository
import com.spaceship.crm.entity.User

interface UserRepository : CrudRepository<User, Long> {
    fun findByUsername(username: String): User?
}